Apeer/fiji_scripts/fijipytools.py

```diff
@@ -477,8 +468,10 @@ def apply_filter(imp, radius=5, filtertype='MEDIAN'):
             ip = stack.getProcessor(index)
 
             # apply filter based on filtertype
-            # if filtertype == 'MEDIAN':
-            filter.rank(ip, radius, filterdict[filtertype])
+            if filtertype in filterdict:
+                filter.rank(ip, radius, filterdict[filtertype])
+            else:
+                print(f"Argument 'filtertype': {filtertype} not found")
```

Comment:
That is a bit critical I would say. In the original version, you throw with a KeyError or something like that, which is pretty much the same as what you are printing in the `else` statement. However now, you silently ignore the error (except for this print statement, but this does not stop control flow and the caller might not even notice since it goes to stdout) and you might get subtle errors later in the code.

My suggestion: Either leave it as it is or do something meaningful, i.e. [only catch what you can/will handle](https://stackoverflow.com/questions/4673860/what-does-only-catch-exceptions-you-can-handle-really-mean).

Reply:
Modified as suggested